services/kubeapi.go

```diff
@@ -17,6 +18,33 @@ func runKubeAPI(host hosts.Host, etcdHosts []hosts.Host, kubeAPIService v1.KubeA
 	return docker.DoRunContainer(host.DClient, imageCfg, hostCfg, KubeAPIContainerName, host.AdvertisedHostname, ControlRole)
 }
 
+func upgradeKubeAPI(host hosts.Host, etcdHosts []hosts.Host, kubeAPIService v1.KubeAPIService) error {
+	logrus.Debugf("[upgrade/KubeAPI] Checking for deployed version")
+	containerInspect, err := docker.InspectContainer(host.DClient, host.AdvertisedHostname, KubeAPIContainerName)
```

Comment:
I am wondering if you can use some common function to include inspect/stopping/deploying/removing container for each service and just call it from inside upgrade* functions instead of repeating the code in each service

Reply:
it would be clunky and would take a lot of parameters due to the way we have things structured..There are some code duplication here but easier to manage for now.